tests/integration/steps/integration_steps.go

```diff
@@ -0,0 +1,622 @@
+package steps
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	resty "github.com/go-resty/resty/v2"
+	"github.com/gruntwork-io/terratest/modules/helm"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const RETRY_TIMEOUT = 15 * time.Minute
+const RETRY_INTERVAL = 30 * time.Second
+
+var namespace string
+var repairId string
+var releaseName string
+
+var (
+	Info    = Color("\033[1;33m%s\033[0m")
+	Outline = Color("\033[1;34m%s\033[0m")
+)
+
+func Color(colorString string) func(...interface{}) string {
+	sprint := func(args ...interface{}) string {
+		return fmt.Sprintf(colorString,
+			fmt.Sprint(args...))
+	}
+	return sprint
+}
+
+func timeTrack(start time.Time, name string) {
+	elapsed := time.Since(start)
+	log.Println(Info(fmt.Sprintf("%s took %s", name, elapsed)))
+}
+
+func RunShellCommand(command *exec.Cmd) error {
+	err := command.Run()
+	return err
+}
+
+func RunShellCommandAndGetOutput(command *exec.Cmd) string {
+	var outb bytes.Buffer
+	command.Stdout = &outb
+	err := command.Run()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return string(outb.String())
+}
+
+// Find returns true if val exists in the slice array, false otherwise
+func Find(slice []string, val string) bool {
+	for _, item := range slice {
+		if item == val {
+			return true
+		}
+	}
+	return false
+}
+
+func g(t *testing.T) *gomega.WithT {
+	return gomega.NewWithT(t)
+}
+
+func deployCluster(t *testing.T, customValues string, helmValues map[string]string) {
+	clusterChartPath, err := filepath.Abs("../../charts/k8ssandra")
+	g(t).Expect(err).To(gomega.BeNil())
+
+	customChartPath, err := filepath.Abs("charts/" + customValues)
+	g(t).Expect(err).To(gomega.BeNil())
+
+	helmOptions := &helm.Options{}
+
+	helmOptions = &helm.Options{
+		// Enable traefik to allow redirections for testing
+		SetValues:      helmValues,
+		KubectlOptions: k8s.NewKubectlOptions("", "", namespace),
```

Comment:
A couple of observations, possible for a future PR, but nonetheless.

1) Do we need to start thinking about a test `context` naming here for our integration tests w/ KubectlOptions?  I'm thinking about things like having concurrently running jobs that have the same namespace, issues with using an empty string, etc.

2) In a prior effort, I was building out a set of test structs/models for managing some of the redundancy in test setup and configuration.

In this example, I was allowing the test setup to prepare these values.  This also included an optional set of string arguments when needed to apply or create with overrides.  

```golang
type OptionsContext struct {
	HelmOptions *helm.Options
	KubeOptions *k8s.KubectlOptions
	Namespace   string
	ReleaseName string
	Args        []string
}
```
Since this is Go, I suspect we could still consider some of this for adoption.  

IMO, structs allows us to reduce some verbosity and build things more consistently.  In fact, [generics in Golang](https://blog.golang.org/generics-next-step) may be a thing in the near future, so that would blend well with this approach moving forward.  

Just a thought, not show stopper.